integration_tests/klog_test.go

```diff
@@ -306,3 +323,30 @@ func withTmpDir(t *testing.T, f func(string)) {
 
 	f(tmpDir)
 }
+
+// getLogFileFromDir returns the path of either the symbolic link to the logfile, or the the logfile itself. This must
+// be done as the creation of a symlink is not guaranteed on any platform. On Windows, only users with administration
+// privileges can create a symlink.
+func getLogFilePath(dir, binaryName, levelName string) (string, error) {
+	symlink := filepath.Join(dir, binaryName+"."+levelName)
+	if _, err := os.Stat(symlink); err == nil {
+		return symlink, nil
+	}
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return "", fmt.Errorf("could not read directory %s: %v", dir, err)
+	}
+	var foundFile string
+	for _, file := range files {
+		if strings.HasPrefix(file.Name(), binaryName) && strings.Contains(file.Name(), levelName) {
+			if foundFile != "" {
+				return "", fmt.Errorf("found multiple matching files")
```

Comment:
shouldn't we be matching the filename exactly instead of matching by HasPrefix and Contains and accounting for multiple matches?

i.e. `if file == binaryName {`

Reply:
I dont think that would be possible. The filename is generated like this: https://github.com/kubernetes/klog/blob/4cae117fdf5d65aeb3f638b8d8c87cd1cf8c8d41/klog_file.go#L97
```
	name = fmt.Sprintf("%s.%s.%s.log.%s.%04d%02d%02d-%02d%02d%02d.%d",
		program,
		host,
		userName,
		tag,
		t.Year(),
		t.Month(),
		t.Day(),
		t.Hour(),
		t.Minute(),
		t.Second(),
		pid)
```

The best we could do would be constructing a larger prefix `program.host.userName.logLevel`. That way the `strings.Contains()` call can be removed. That would come at the cost of having to re-implement the`[shortHostname ](https://github.com/kubernetes/klog/blob/4cae117fdf5d65aeb3f638b8d8c87cd1cf8c8d41/klog_file.go#L88) function within the  test file though.

I'm not sure what would be preferable.

Reply:
hm, yet the original code matched a filename exactly using `assertFileIsAbsent` and `logfile := filepath.Join(logdir, file)`

i'm trying to understand why this is a problem on windows.

other options:
- wouldn't `withTmpDir` ensure that the temp directory where the logs are stored contain only a single file matching the level?
- looks like the filename ends with a pid, can that be used for match too?

Reply:
>wouldn't withTmpDir ensure that the temp directory where the logs are stored contain only a single file matching the level?
Under properly working circumstances yes it should. I included the check on the off chance that something went wrong, like multiple log files for the same level being used.

Reply:
ok let's leave this as is for now, unless someone else objects.